pkg/querier/queryrange/results_cache.go

```diff
@@ -49,43 +54,51 @@ func (cfg *ResultsCacheConfig) RegisterFlags(f *flag.FlagSet) {
 type Extractor interface {
 	// Extract extracts a subset of a response from the `start` and `end` timestamps in milliseconds in the `from` response.
 	Extract(start, end int64, from Response) Response
+	ExtractCacheGenNumber(from Response) string
```

Comment:
If we are going to get the Cache gen number here, can we propagate it to the downstream calls using a context value, instead of retrieving for each call to the store/cache?